### PR TITLE
Chore: Create tests for http client

### PR DESCRIPTION
**Pull request by RLRabinowitz**
### Issue & Steps to Reproduce / Feature Request
We want UTs for our http client

### Solution
Reverse the dependency of the RestClient
Then, mock http and create tests

### CODE REVIEW

File: client/aws_credentials.go [hunk 1]
```diff
@@ -53,4 +53,4 @@ func (self *ApiClient) AwsCredentialsCreate(request AwsCredentialsCreatePayload)
 
 func (self *ApiClient) AwsCredentialsDelete(id string) error {
 	return self.http.Delete("/credentials/" + id)
-}
\ No newline at end of file
+}
```
Comment (RLRabinowitz) [inline client/aws_credentials.go]: Ran `./build.sh`...

File: client/aws_credentials_test.go [hunk 2]
```diff
@@ -15,14 +15,14 @@ var _ = Describe("AwsCredentials", func() {
 		Id:             "id1",
```
Comment (RLRabinowitz) [inline client/aws_credentials_test.go:15]: All these file changes are because I ran `./build.sh`...

File: client/project.go [hunk 3]
```diff
@@ -43,7 +43,7 @@ func (self *ApiClient) ProjectDelete(id string) error {
 
 func (self *ApiClient) ProjectUpdate(id string, payload UpdateProjectPayload) (Project, error) {
 	var result Project
-	err := self.http.Put("/projects/" + id, payload, &result)
+	err := self.http.Put("/projects/"+id, payload, &result)
```
Comment (RLRabinowitz) [inline client/project.go:46]: `go fmt`...

File: client/project_test.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ var _ = Describe("Project", func() {
 	mockProject := Project{
```
Comment (RLRabinowitz) [inline client/project_test.go:14]: All of this file is `go fmt`...

File: env0/resource_aws_credentials.go [hunk 5]
```diff
@@ -46,9 +46,9 @@ func resourceAwsCredentials() *schema.Resource {
 func resourceAwsCredentialsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (RLRabinowitz) [inline env0/resource_aws_credentials.go:46]: `go fmt`...

File: env0/provider.go [hunk 6]
```diff
@@ -62,7 +63,8 @@ func configureProvider(d *schema.ResourceData) (interface{}, error) {
 		return nil, errors.New("either api_secret must be provided or ENV0_API_SECRET environment variable set")
 	}
 
-	httpClient, err := http.NewHttpClient(apiKey.(string), apiSecret.(string), d.Get("api_endpoint").(string))
+	restClient := resty.New()
+	httpClient, err := http.NewHttpClient(apiKey.(string), apiSecret.(string), d.Get("api_endpoint").(string), restClient)
```
Comment (yaronya) [inline env0/provider.go:67]:
```suggestion
      httpClient, err := http.NewHttpClient(apiKey.(string), apiSecret.(string), d.Get("api_endpoint").(string), resty.New())

File: client/http/client_test.go [hunk 7]
```diff
@@ -0,0 +1,220 @@
+package http_test
+
+import (
+	"encoding/json"
+	httpkaki "github.com/env0/terraform-provider-env0/client/http"
```
Comment (yaronya) [inline client/http/client_test.go]: tf?

File: client/http/client_test.go [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+})
+
+var _ = AfterSuite(func() {
+	// unmock HTTP requests
+	httpmock.DeactivateAndReset()
+})
+
+func TestHttpClient(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "API Client Tests")
```
Comment (yaronya) [inline client/http/client_test.go:45]:
```suggestion
	RunSpecs(t, "HTTP Client Tests")

File: client/http/client_test.go [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+// The JSON here returns camelCase keys
+type ResponseType struct {
```
Comment (yaronya) [inline client/http/client_test.go:49]: could be nice if you can hoist these types like we do in TS

**Review by yaronya — APPROVED** [review]
Well done! 💯 
Left some small comments 🙏

**Comment by RLRabinowitz** [comment]
[UT for client/http/client.go](https://trello.com/c/N2jhuKb8/3371-ut-for-client-http-clientgo)

File: client/http/client_test.go (hunk 7 above)
Comment (RLRabinowitz) [reply client/http/client_test.go]: Woops, changed in https://env0.slack.com/archives/C019V93M48P/p1622556211046200

**Pull request merged**